modules/picard/fastqtosam/main.nf

```diff
@@ -0,0 +1,45 @@
+process PICARD_FASTQTOSAM {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::picard=2.27.4" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/picard:2.27.4--hdfd78af_0' :
+        'quay.io/biocontainers/picard:2.27.4--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[Picard FastqToSam] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+    def input = meta.single_end ? "--FASTQ ${reads}" : "--FASTQ ${reads[0]} --FASTQ2 ${reads[1]}"
+    def sample_name = args.contains("--SAMPLE_NAME") || args.contains("-SM") ? "" : "--SAMPLE_NAME ${prefix}"
```

Comment:
This doensn't seem to do anything 😞

Reply:
The custom samplename testcase just ignores this line it seems

Reply:
Why don't you just add the prefix in the `args` itself? like `ext.args = {"--SAMPLE_NAME $prefix"}` instead of `ext.args = "--SAMPLE_NAME CustomSample"` and adjusting it in the module itself?

Reply:
`--SAMPLE_NAME` is a mandatory argument, so we need to catch the case where the user did not set an explicit sample name in `ext.args`

Reply:
Allright, looks good to me then. Is it working now?

Reply:
seems like it 👍🏻